Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CertificateStateHelper.kt

```diff
@@ -0,0 +1,140 @@
+                expirationStatusIcon.setImageDrawable(context.getDrawableCompat(R.drawable.ic_av_timer))
+                expirationStatusText.visibility = View.VISIBLE
+                expirationStatusText.text = context.getString(
+                    R.string.certificate_qr_expiration,
+                    certificate.headerExpiresAt.toShortDayFormat(),
+                    certificate.headerExpiresAt.toShortTimeFormat()
+                )
+                expirationStatusBody.visibility = setVisibility(isCertificateDetails)
+                expirationStatusBody.text = context.getText(R.string.expiration_info)
+                startValidationCheckButton.visibility = setVisibility(isPersonDetails)
+            }
+
+            is CwaCovidCertificate.State.Expired -> {
+                expirationStatusIcon.visibility = View.VISIBLE
+                expirationStatusIcon.setImageDrawable(context.getDrawableCompat(R.drawable.ic_error_outline))
+                expirationStatusText.visibility = View.VISIBLE
+                expirationStatusText.text = context.getText(R.string.certificate_qr_expired)
+                expirationStatusBody.visibility = setVisibility(isCertificateDetails)
+                expirationStatusBody.text = context.getText(R.string.expired_certificate_info)
+                qrSubtitle.visibility = View.GONE
+                startValidationCheckButton.visibility = setVisibility(isPersonDetails)
+            }
+
+            is CwaCovidCertificate.State.Invalid -> {
+                expirationStatusIcon.visibility = View.VISIBLE
+                expirationStatusIcon.setImageDrawable(context.getDrawableCompat(R.drawable.ic_error_outline))
+                expirationStatusText.visibility = View.VISIBLE
+                expirationStatusText.text = context.getText(R.string.certificate_qr_invalid_signature)
+                expirationStatusBody.visibility = setVisibility(isCertificateDetails)
+                expirationStatusBody.text = context.getText(R.string.invalid_certificate_signature_info)
+                qrSubtitle.visibility = View.GONE
+                startValidationCheckButton.visibility = setVisibility(isPersonDetails)
+            }
+
+            is CwaCovidCertificate.State.Valid -> {
+                expirationStatusIcon.visibility = View.GONE
+                expirationStatusText.visibility = View.GONE
+                expirationStatusBody.visibility = View.GONE
+                qrTitle.visibility = setVisibility(isPersonDetails)
+                qrSubtitle.visibility = setVisibility(isPersonDetails)
+                startValidationCheckButton.visibility = setVisibility(isPersonDetails)
+            }
+        }
+    }
+
+    private fun setVisibility(visibility: Boolean) =
+        if (visibility) {
+            View.VISIBLE
+        } else {
+            View.GONE
+        }
+
+    fun displayIndividualCardsExpirationState(
+        certificateExpiration: TextView,
+        context: Context,
```

Comment:
you can access the context from the view  😉

Reply:
Good point :)